src/components/drawing/index.js

```diff
@@ -426,6 +425,35 @@ drawing.pointStyle = function(s, trace, gd) {
     });
 };
 
+drawing.selectedPointStyle = function(s, trace) {
+    if(!s.size() || !trace.selectedpoints) return;
+
+    var selectedAttrs = trace.selected || {};
+    var unselectedAttrs = trace.unselected || {};
+
+    s.style('opacity', function(d) {
+        return d.selected ?
+            (selectedAttrs.marker || {}).opacity :
+            (unselectedAttrs.marker || {}).opacity;
+    });
+
+    // which is slightly different than:
+    // ((d.mo + 1 || opacity + 1) - 1) * (d.dim ? DESELECTDIM : 1);
+    // in https://github.com/plotly/plotly.js/blob/master/src/traces/scatter/select.js
```

Comment:
This means that if marker opacity is an array, you won't be able to keep that during selection. I feel like we can do better than that:
- Only give `[un]selected.marker.opacity` a default value if you don't set any other `[un]selected` attributes - so if you want to keep opacity intact but just change marker color during selection, you can do that.
- If only one of the `[un]selected` opacities is set, the other state should revert to the `arrayOk` logic `(d.mo + 1 || marker.opacity + 1) - 1`
- The same is true for colors: if only one is used and the base attribute is an array, the other state should revert to the `arrayOk` logic - perhaps we want to stash this color in `d` though, as this logic is a bit more involved? Anyway, that would allow selections like in parcoords (deselected is gray, selected keeps its color) or like our S&P notifiers (deselected keeps its color, selected is bright red)
- And for both opacity and color, if neither is set you don't even need to do that loop.
- Another minor 🐎  - this will probably naturally happen with the above, but you can move `(selectedAttrs.marker || {}).opacity` calculation out of the `.each` loop - also `smc` and `usmc` (Semper Fi!) below. That said I'd love it if some day these attributes all become `arrayOk` in which case they'd have to move at least partially back into the loop... But that's not for this PR, single-value is great for now.
- My suggestions may make a mess with gradients... perhaps for now if you have a gradient and only one selection color we just revert to uniform `marker.color` for the other state?

Reply:
> Only give [un]selected.marker.opacity a default value if you don't set any other [un]selected attributes - so if you want to keep opacity intact but just change marker color during selection, you can do that.

But then `[un]selected.marker.opacity` won't appear in `_fullData`, and @monfera :smirk: won't be able to populate his style panels. Personally, I think keeping the same default selection behavior is more important, so thanks @alexcjohnson for pointing this out!

> My suggestions may make a mess with gradients... perhaps for now if you have a gradient and only one selection color we just revert to uniform marker.color for the other state?

Good point here. I haven't tested any of the mocks with `marker.gradient` turned on.

Reply:
Revised logic in 1859256 and d3cdd6f with tests.